staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go

```diff
@@ -126,16 +127,34 @@ func ValidateNoNewFinalizers(newFinalizers []string, oldFinalizers []string, fld
 	}
 	return allErrs
 }
-
-// ValidateImmutableField validates the new value and the old value are deeply equal.
-func ValidateImmutableField(newVal, oldVal interface{}, fldPath *field.Path) field.ErrorList {
+func ValidateImmutableFieldWithErrMsg(newVal, oldVal interface{}, errMsg string, fldPath *field.Path) field.ErrorList {
```

Comment:
Can we change this method signature to take a `field.Error`, so that callers can call directly: `field.Invalid` or `field.Forbidden` depending on their current use-case. Then this method can change the `details` of the `field.Error` with result from cmp.Diff result?